jupyterlab_server/settings_handler.py

```diff
@@ -242,19 +242,21 @@ def put(self, schema_name):
         if not settings_dir:
             raise web.HTTPError(500, settings_error)
 
-        raw = self.request.body.strip().decode(u'utf-8')
+        raw_payload = self.request.body.strip().decode(u'utf-8')
+        raw_settings = json.loads(raw_payload)['raw']
+        payload = json5.loads(raw_settings)
```

Comment:
Since we are parsing user input - do you think we should wrap this in a try/except that raises a http 400 error like below? That would give user feedback, rather than just erroring out here with a server error.

Reply:
Sounds good, I can do that :)

Reply:
Not only did I change it to have exception handling, I also added tests for them :)